### PR TITLE
[ADF-4936] Auth guard redirect fix

### PR DESCRIPTION
**Pull request by suzanadirla**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ADF-4936

**What is the new behaviour?**
Allow the AuthGuard to redirect the user when **needed**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/services/auth-guard.service.ts [hunk 1]
```diff
@@ -73,6 +73,9 @@ export class AuthGuard extends AuthGuardBase {
         if (!this.authenticationService.isOauth() || this.isOAuthWithoutSilentLogin()) {
             this.redirectToUrl('ALL', redirectUrl);
         }
+        if (this.isOAuthWithSilentLogin()) {
```
Comment (adinapitul) [inline lib/core/services/auth-guard.service.ts]: should this be added for auth-guard-ecm and auth-guard-bpm as well?

File: lib/core/services/auth-guard-base.ts [hunk 2]
```diff
@@ -91,4 +91,14 @@ export abstract class AuthGuardBase implements CanActivate, CanActivateChild {
             this.authenticationService.isOauth() && oauth.silentLogin === false
         );
     }
+
+    protected isOAuthWithSilentLogin() {
```
Comment (DenysVuika) [inline lib/core/services/auth-guard-base.ts]: missing return type

File: lib/core/services/auth-guard-base.ts [hunk 3]
```diff
@@ -91,4 +91,14 @@ export abstract class AuthGuardBase implements CanActivate, CanActivateChild {
...
+        const oauth = this.appConfigService.get<OauthConfigModel>(
+            AppConfigValues.OAUTHCONFIG,
+            null
+        );
+        return (
```
Comment (DenysVuika) [inline lib/core/services/auth-guard-base.ts]: you get `AppConfigValues.OAUTHCONFIG` or `null`. missing the null check in the code below, may crash at runtime

**Comment by suzanadirla** [comment]
Reopened this PR as these changes are needed to fix this issue https://issues.alfresco.com/jira/browse/ADF-4936 (With SSO enabled, blank empty page displayed when route is matched as public but is wrong)

**Comment by eromano** [comment]
the implicit login concern is already in the JS-API

**Comment by suzanadirla** [comment]
ok, I removed the implicit login logic and used the redirectToUrl instead. @eromano please review

**Pull request merged**